pkg/kritis/util/util.go

```diff
@@ -36,16 +60,94 @@ func GetResource(image string) *grafeas.Resource {
 	return &grafeas.Resource{Uri: GetResourceURL(image)}
 }
 
-func CreateAttestationSignature(image string, pgpSigningKey *secrets.PGPSigningSecret) (string, error) {
-	hostSig, err := container.NewAtomicContainerSig(image, map[string]string{})
+func CreateAttestation(image string, pgpSigningKey *secrets.PGPSigningSecret) (*cryptolib.Attestation, error) {
+	hostSig, err := container.NewAtomicContainerSig(image, nil)
+	if err != nil {
+		return nil, err
+	}
+	payload, err := hostSig.JSON()
+	if err != nil {
+		return nil, err
+	}
+
+	privateKey, err := extractPrivateKeyBytes(pgpSigningKey)
+	if err != nil {
+		return nil, err
+	}
+	signer, err := cryptolib.NewPgpSigner(privateKey)
+	if err != nil {
+		return nil, err
+	}
+
+	att, err := signer.CreateAttestation([]byte(payload))
+	if err != nil {
+		return nil, err
+	}
+	return att, nil
+}
+
+func extractPrivateKeyBytes(secret *secrets.PGPSigningSecret) ([]byte, error) {
+	keyBuffer := bytes.NewBuffer(nil)
+	armorWriter, err := armor.Encode(keyBuffer, openpgp.PrivateKeyType, nil)
+	if err != nil {
+		return nil, err
+	}
+	entity, err := createEntityFromKeys(secret.PgpKey.PublicKey(), secret.PgpKey.PrivateKey())
 	if err != nil {
-		return "", err
+		return nil, err
 	}
-	hostStr, err := hostSig.JSON()
+	err = entity.SerializePrivate(armorWriter, nil)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
-	return attestation.CreateMessageAttestation(pgpSigningKey.PgpKey, hostStr)
+	armorWriter.Close()
+	return keyBuffer.Bytes(), nil
+}
+
+func createEntityFromKeys(pubKey *packet.PublicKey, privKey *packet.PrivateKey) (*openpgp.Entity, error) {
+	currentTime := pgpConfig.Now()
+	uid := packet.NewUserId("", "", "")
+	if uid == nil {
+		return nil, errors.New("user id field contained invalid characters")
+	}
```

Comment:
This error does not make sense because NewUserId is taking no arguments

Reply:
Done.